packages/frontend/src/redux/slices/tokens/index.js

```diff
@@ -78,26 +82,32 @@ const fetchTokens = createAsyncThunk(
                     }));
                 }
 
-                const config = {
+                const tokenConfig = {
                     contractName,
-                    data: {
-                        contractName,
-                        balance,
-                        onChainFTMetadata,
-                        fiatValueMetadata: tokenFiatValues.tokens[contractName] || {},
-                    },
+                    balance,
+                    onChainFTMetadata,
+                    fiatValueMetadata: tokenFiatValues.tokens[contractName] || {},
                 };
 
+                tokens[contractName] = tokenConfig;
+
                 if (Number(balance)) {
-                    dispatch(addTokenWithBalance(config));
-                } else {
-                    dispatch(addToken(config));
+                    tokensWithBalance[contractName] = tokenConfig;
                 }
             } catch (e) {
                 // Continue loading other likely contracts on failures
                 console.warn(`Failed to load FT for ${contractName}`, e);
             }
         }));
+
+        batch(() => {
+            dispatch(setTokens(tokens));
+            dispatch(
+                setTokensWithBalance(
+                    sortTokensInDecreasingOrderByPrice(tokensWithBalance)
```

Comment:
I feel like "sorting" tokens is purely presentational logic, so it should not live in redux, but instead somewhere closer to the components.
Maybe a hook `const sortedTokens = useSortedTokens(tokens, (a, b) => sortTokensByPrice(a, b, prices))`